index.js

```diff
@@ -88,28 +77,9 @@ app.get("/persons/:name", (req, res) => {
   }
 });
 
-app.delete("/persons/:name", (req, res) => {
-  const nameReq = req.params.name || "";
-  const deletedPerson = persons.filter((person) => person.name === nameReq);
-  const filteredPersons = persons.filter((person) => person.name !== nameReq);
-
-  if (deletedPerson.length > 0) {
-    res.status(202).send(`user: ${nameReq} was deleted`);
-    persons = filteredPersons;
-    updatePersons.savePersons(persons);
-    return;
-  }
-
-  if (deletedPerson.length <= 0) {
-    res.status(204).send(`${nameReq} person not found`);
-    return;
-  }
-
-  res.status(400).send(`<h1>400</h1><br /> <p>Bad request</p>`);
-});
-
 app.put("/persons/:name", (req, res) => {
   const { query } = req;
+  const queryExists = Object.keys(query).length > 0
```

Comment:
Why do you need query param of the identifier of the person is already part of the path?

Reply:
in the PUT endpoint I have 3 cases
the first is accepted(the object exist + query to update exist)
the second is bad query(the object exist but there's no query)
the last is not found(object doesn't exist)

it checks if the param also come with queries

without this I couldn't test for the first case, it was ignoring the second case and going directly to the thrid case
so i made this helper const to fix this